alf/environments/igibson/igibson_tasks.py

```diff
@@ -0,0 +1,816 @@
+# Copyright (c) 2021 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import igibson
+from igibson.tasks.task_base import BaseTask
+from igibson.scenes.igibson_indoor_scene import InteractiveIndoorScene
+from igibson.scenes.gibson_indoor_scene import StaticIndoorScene
+from igibson.termination_conditions.max_collision import MaxCollision
+from igibson.termination_conditions.timeout import Timeout
+from igibson.termination_conditions.out_of_bound import OutOfBound
+from igibson.termination_conditions.point_goal import PointGoal
+from igibson.reward_functions.potential_reward import PotentialReward
+from igibson.reward_functions.collision_reward import CollisionReward
+from igibson.reward_functions.point_goal_reward import PointGoalReward
+from igibson.utils.utils import l2_distance, rotate_vector_3d, cartesian_to_polar
+from igibson.objects.visual_marker import VisualMarker
+
+import numpy as np
+import os
+import time
+import logging
+import pybullet as p
+
+from alf.environments.igibson.igibson_object import iGibsonObject
+
+
+class VisualPointNavFixedTask(BaseTask):
```

Comment:
It's better to derive this class from PointNavFixedTask because most functions have already been implemented there, so it's unnecessary to redo the work here.